btree/btree.c

```diff
@@ -4845,16 +4867,24 @@ static int64_t root_case_handle(struct m0_btree_op *bop)
 		struct slot     root_slot = {};
 		struct segaddr  root_child;
 		struct level   *root_lev = &oi->i_level[0];
-		/* TBD: check if node_lock or node_count check is needed. */
+
+		node_lock(root_lev->l_node);
+
+		if (!node_isvalid(root_lev->l_node) ||
+		    oi->i_level[0].l_seq != root_lev->l_seq) {
```

Comment:
are you expecting that l_seq might change between line 4869 and line 4870?

Reply:
check is incorrect, It should be root_lev->l_node->n_seq != root_lev->l_seq. I will make changes.